### DISCUSSION TITLE
Fragments?

### DISCUSSION THREAD

**Discussion by matthewlilley** [graphprotocol/graph-client#331] [Q&A]
I have cases where I might want to call the same entity multiple times using the timetravel feature of graph-node. This is useful for rolling windows of data, like 24h volume, 24h fees, 24h transaction count etc... Currently this will select EVERYTHING which the base entity needs, but actually only needs a few fields per entity. 

First though was fragments, but unsure what that looks like here...

An example use case:
https://github.com/sushiswap/sushiswap/blob/refactor/graph-client/packages/graph-client-2/resolvers/pairs/index.ts#L54

**Comment by saihaj** [comment]
I am not following your question here. You should be a let of use fragments in your queries.

**Reply by matthewlilley** [reply]
@saihaj an example https://github.com/sushiswap/sushiswap/blob/master/packages/graph-client/resolvers/pairs/index.ts#L45

The queries for pools1d, pools2d, pools1w use the pairs query but only need a limited number of selections.
